calendar_reboot.js

```diff
@@ -161,17 +188,13 @@ $(document).ready(function() {
         // and fill it with the information given in the form
         //required info are startDate and title. default to tracking 1 
         //year
+        
         var calendarTitle = $("#calendarTitle").val();
         var startDate = $("#startDate").val();
         var endDate = $("#endDate").val();
-            
         
-        var validateTitle = validateInput(calendarTitleId);
-        var validateStartDate = validateInput(startDateId);
-        var validateEndDate = validateInput(endDateId);
-        var validateDates = validateForm(startDate, endDate);
+        if (validateForm() && validateDates(startDate, endDate)) {
```

Comment:
So, I included validateDates in the validateForm function. The validateForm function first checks that input was actually given. When the user has filled out all three fields, it then checks if the dates are valid (whether end date occurs after start date). So now in the set button click handler you only have to check if validateForm is true and then you can build your calendar.